src/ast/passes/codegen_llvm.cpp

```diff
@@ -4697,4 +4710,37 @@ CallInst *CodegenLLVM::CreateKernelFuncCall(Kfunc kfunc,
   return b_.createCall(func->getFunctionType(), func, args, name);
 }
 
+/// This should emit
+///
+///    declare !dbg !... extern ... @var_name(...) section ".ksyms"
+///
+/// with proper debug info entry.
+///
+/// The function type is retrieved from kernel BTF.
+///
+/// If the function declaration is already in the module, just return it.
+///
+GlobalVariable *CodegenLLVM::DeclareKernelVar(const std::string &var_name)
+{
+  if (auto *sym = module_->getGlobalVariable(var_name))
+    return sym;
+
+  std::string err;
+  auto type = bpftrace_.btf_->get_var_type(var_name);
+  if (type.IsNoneTy()) {
+    throw FatalUserException("Could not resolve variable \"" + var_name +
+                             "\" from BTF");
+  }
+
```

Comment:
Can this be done in semantic analyser so there's highlighting?

Reply:
Right, makes sense.